pkg/apis/autoscaling/v2/defaults.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/apis/autoscaling"
+)
+
+var (
+	// These constants repeats previous HPA behavior
+	scaleUpLimitPercent         int32 = 100
+	scaleUpLimitMinimumPods     int32 = 4
+	scaleUpPeriod               int32 = 15
+	scaleUpStabilizationSeconds int32
+	maxPolicy                   = autoscalingv2.MaxChangePolicySelect
```

Comment:
Per previous comments, I strongly prefer "MaxChange" and "MinChange" to "Max" and "Min".  This feature (scale controls) doesn't yet have wide adoption and the existing values are confusing for those have have.

This is the *only* change from v2beta2 that I feel strongly about.

Reply:
The current state of this PR only changes the go field name in v2, not the serialized value. That means conversion and validation don't have to change.

Making the serialized enum value be different between v2beta2 and v2 means custom conversion is required, and validation has to be made version-aware so it can return the correct message about what values are allowed. That is non-trivial to do... do we think the change is worth it?

If so, this PR would have to change to this, and add value-conversion and version-aware validation:

```diff
diff --git a/staging/src/k8s.io/api/autoscaling/v2/types.go b/staging/src/k8s.io/api/autoscaling/v2/types.go
index 910690b4a27..315f88c2dbe 100644
--- a/staging/src/k8s.io/api/autoscaling/v2/types.go
+++ b/staging/src/k8s.io/api/autoscaling/v2/types.go
@@ -156,9 +156,9 @@ type ScalingPolicySelect string
 
 const (
 	// MaxChangePolicySelect  selects the policy with the highest possible change.
-	MaxChangePolicySelect ScalingPolicySelect = "Max"
+	MaxChangePolicySelect ScalingPolicySelect = "MaxChange"
 	// MinChangePolicySelect selects the policy with the lowest possible change.
-	MinChangePolicySelect ScalingPolicySelect = "Min"
+	MinChangePolicySelect ScalingPolicySelect = "MinChange"
 	// DisabledPolicySelect disables the scaling in this direction.
 	DisabledPolicySelect ScalingPolicySelect = "Disabled"
 )
```

The following test cases would be required:
1. make sure `MaxChange` and `MinChange` are not permitted via create or update requests to v1, v2beta1, or v2beta2 endpoints, and the returned validation error indicates the valid values are `Max` and `Min`
2. make sure `Max` and `Min` are not permitted via create or update requests to v2 endpoints, and the returned validation error indicates the valid values are `MaxChange` and `MinChange`
3. make sure an HPA created with `Max` and `Min` via v1, v2beta1, and v2beta2 endpoints returns `MaxChange` and `MinChange` when read via the `v2` endpoint
4. make sure an HPA created with `MaxChange` and `MinChange` via the v2 endpoint returns `Max` and `Min` when read via the `v1`, `v2beta1` and `v2beta2` endpoints

Reply:
also, it's not clear what conversion should do to distinguish between values of `Max` and `MaxChange` sent to the v2 API, in order to prevent values of `Max` from being accepted via the v2 API

Reply:
I think this works:
1. external → internal conversion should not translate values
2. internal validation needs to become aware of the API version the request is being validated for
   * when validating v1, v2beta1, v2beta2: use the current `validSelectPolicyTypes` list for validation and error messages
   * when validating any other version: use a new list of `MaxChange`, `MinChange`, `Disabled` for validation and error messages
3. internal → external conversion should translate enum values to the accepted value for that version
   * internal → {v1, v2beta1, v2beta2}: MaxChange → Max, MinChange → Min
   * internal → any other version: Max → MaxChange, Min → MinChange

Reply:
see `ValidatePodDisruptionBudgetStatusUpdate` and `podDisruptionBudgetStatusStrategy#ValidateUpdate` for an example of plumbing the API version to a validation function

Reply:
So, I will  rename go field name(MaxPolicySelect->MaxChangePolicySelect , MinPolicySelect->MaxChangePolicySelect) with keeping the values( Max and Min).

Reply:
@josephburnett, does that match your expectations?

Reply:
> That is non-trivial to do... do we think the change is worth it?

@liggitt seeing how much difficulty it is, no I don't think it is worth changing the values as well.  @wangyysde you suggest to rename the fields but not the values.  That's a good middle ground so it will be more clear.  Let's go with that.